signal/signal.go

```diff
@@ -14,49 +14,64 @@ import (
 )
 
 var (
-	// interruptChannel is used to receive SIGINT (Ctrl+C) signals.
-	interruptChannel = make(chan os.Signal, 1)
-
-	// shutdownRequestChannel is used to request the daemon to shutdown
-	// gracefully, similar to when receiving SIGINT.
-	shutdownRequestChannel = make(chan struct{})
-
 	// started indicates whether we have started our main interrupt handler.
 	// This field should be used atomically.
 	started int32
+)
 
-	// quit is closed when instructing the main interrupt handler to exit.
-	quit = make(chan struct{})
+// Interceptor contains channels and methods regarding application shutdown
+// and interrupt signals
+type Interceptor struct {
+	// InterruptChannel is used to receive SIGINT (Ctrl+C) signals.
+	InterruptChannel chan os.Signal
 
-	// shutdownChannel is closed once the main interrupt handler exits.
-	shutdownChannel = make(chan struct{})
-)
+	// ShutdownChannel is closed once the main interrupt handler exits.
+	ShutdownChannel chan struct{}
```

Comment:
I think this is the only channel that needs to be exported here?

Reply:
We can even keep the `ShutdownChannel()` function and keep everything in this struct private.

Reply:
Definitely! Fixed.